librad/src/identities/git.rs

```diff
@@ -174,8 +175,33 @@ impl<'a, T: 'a> Identities<'a, T> {
         root.verify(progeny)
     }
 
+    pub fn is_fork_generic(&self, left: git2::Oid, right: git2::Oid) -> Result<bool, git2::Error> {
+        if left == right {
+            return Ok(false);
+        }
+
+        Ok(if self.is_in_ancestry_path(left, right)? {
+            false
+        } else {
+            !(self.is_in_ancestry_path(right, left)?)
+        })
+    }
+
     //// Helpers ////
 
+    /// Assumes that the bag of commits are in a related history.
+    fn latest_generic(&self, bag: NonEmpty<git2::Oid>) -> Result<git2::Oid, git2::Error> {
+        let mut latest = bag.head;
+
+        for commit in bag.into_iter() {
+            if self.repo.graph_descendant_of(latest, commit)? {
```

Comment:
I think the argument order of `graph_descendant_of` is assumed-to-be-more-recent, ancestor. So this would find the oldest commit.

Reply:
I want to ban all booleans from programming

Reply:
:white_check_mark: